typesense_derive/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+use quote::quote;
+
+#[proc_macro_derive(TypesenseCollection, attributes(typesense_collection))]
+pub fn typesense_collection_derive(input: TokenStream) -> TokenStream {
+    // TODO: better error handling
```

Comment:
How are you planning to tackle this?

Reply:
I plan to use this:

https://doc.rust-lang.org/std/macro.compile_error.html

https://docs.rs/syn/1.0/syn/struct.Error.html#method.to_compile_error

I still have to refactor a lot this implementation. Move a lot of code to functions and check more things, for instance we need to check the default_sorting_field is a valid field.

I hadn't have done a procedural macro before, I have been following this workshop in the last couple of days https://github.com/dtolnay/proc-macro-workshop/